### ISSUE TITLE
sql: RSG failure

### ISSUE THREAD

**Issue by jordanlewis** [cockroachdb/cockroach#14628]
This looks like a error during CREATE DATABASE if I'm not mistaken. Assigning to @mjibson for initial triage.

```
panic: pq: unexpected value: raw_bytes:"\354`!\240\001\212\001" timestamp:<wall_time:1491372392340357838 logical:0 > 

goroutine 352 [running]:
github.com/cockroachdb/cockroach/pkg/sql_test.TestRandomSyntaxGeneration.func1(0xc420ba59a0, 0xc420b5eb10, 0xc42072d4a0, 0xc4206ae000)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rsg_test.go:67 +0x255
github.com/cockroachdb/cockroach/pkg/sql_test.testRandomSyntax.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rsg_test.go:233 +0xc0
created by github.com/cockroachdb/cockroach/pkg/sql_test.testRandomSyntax
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rsg_test.go:244 +0x558
```

https://teamcity.cockroachdb.com/viewLog.html?buildId=207469&buildTypeId=Cockroach_Nightlies_RandomSyntaxTests&tab=buildLog#_

**Comment by mjibson** [comment]
The unexpected value error occurs in roachpb/errors.go. The raw_bytes thing seems like something produced by the rocksdb code? I'm not sure where to dig here as there's not a lot useful in the error. @petermattis any ideas?

**Comment by petermattis** [comment]
`Value.raw_bytes` (a.k.a. `Value.RawBytes`) is where column data is stored in the KV layer. I'd guess the error is coming from a failed `ConditionalPut`. See `ConditionFailedError`. You could try instrumenting the code which is creating the error and see what it is getting called.

**Comment by mjibson** [comment]
Ok. This is the first time we've seen this bug. I'm planning on waiting for a while to see if it comes up again.

**Comment by petermattis** [comment]
Well, we really shouldn't be leaking that error through to `pq`. I think RSG found a real problem again.

**Comment by BramGruneir** [comment]
This seems related to #13504.  But I've been unable to reproduce those failures.

**Comment by mjibson** [comment]
Closing this, will reopen if it happens again.

**Comment by mjibson** [comment]
Reopening because a user in gitter got:

`F170510 17:35:22.081732 12 server/server.go:851  [n2] failed to run migration "create system.settings table": unexpected value: raw_bytes:"\306DY\334\001\014" timestamp:<wall_time:1494437673208752830 logical:0 >`

**Comment by clanstyles** [comment]
This was me @mjibson - if there's anything I can provide to try to track this down, let me know. All docker shows is:

```
kubectl logs cockroachdb-1
+ CRARGS=("start" "--logtostderr" "--insecure" "--host" "$(hostname -f)" "--http-host" "0.0.0.0")
++ hostname -f
++ hostname
+ '[' '!' cockroachdb-1 == cockroachdb-0 ']'
+ CRARGS+=("--join" "cockroachdb-public")
+ exec /cockroach/cockroach start --logtostderr --insecure --host cockroachdb-1.cockroachdb.production.svc.cluster.local --http-host 0.0.0.0 --join cockroachdb-public
I170510 17:35:21.225937 1 cli/start.go:575  CockroachDB CCL v1.0 (linux amd64, built 2017/05/10 13:19:57, go1.8.1)
I170510 17:35:21.327391 12 cli/start.go:338  starting cockroach node
W170510 17:35:21.328137 12 server/server.go:162  running in insecure mode, this is strongly discouraged. See --insecure.
W170510 17:35:21.336568 12 gossip/gossip.go:1189  [n?] no incoming or outgoing connections
I170510 17:35:21.337127 12 storage/engine/rocksdb.go:380  opening rocksdb instance at "/cockroach/cockroach-data"
I170510 17:35:21.360480 33 gossip/client.go:131  [n?] started gossip client to cockroachdb-public:26257
I170510 17:35:21.397495 12 server/config.go:421  1 storage engine initialized
I170510 17:35:21.397894 12 server/server.go:678  [n?] sleeping for 435.91903ms to guarantee HLC monotonicity
I170510 17:35:21.835134 12 storage/store.go:1190  [n2] [n2,s2,r3/?:/Table/1{1-2}]: added to replica GC queue
I170510 17:35:21.835578 12 storage/store.go:1190  [n2] [n2,s2,r7/?:/{Table/15-Max}]: added to replica GC queue
I170510 17:35:21.835896 12 server/node.go:467  [n2] initialized store [n2,s2]: {Capacity:1023303680 Available:950136832 RangeCount:2 LeaseCount:0}
I170510 17:35:21.836026 12 server/node.go:351  [n2] node ID 2 initialized
I170510 17:35:21.836295 12 gossip/gossip.go:297  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"cockroachdb-1.cockroachdb.production.svc.cluster.local:26257" > attrs:<> locality:<> 
I170510 17:35:21.836503 12 storage/stores.go:296  [n2] read 2 node addresses from persistent storage
I170510 17:35:21.836819 12 server/node.go:608  [n2] connecting to gossip network to verify cluster ID...
I170510 17:35:21.836962 12 server/node.go:633  [n2] node connected via gossip and verified as part of cluster "28123fc5-7526-4ff4-a744-fc2026930889"
I170510 17:35:21.837122 12 server/node.go:405  [n2] node=2: started with [[]=/cockroach/cockroach-data] engine(s) and attributes []
I170510 17:35:21.837319 12 sql/executor.go:338  [n2] creating distSQLPlanner with address {tcp cockroachdb-1.cockroachdb.production.svc.cluster.local:26257}
I170510 17:35:21.840614 12 server/server.go:741  [n2] starting http server at 0.0.0.0:8080
I170510 17:35:21.840829 12 server/server.go:742  [n2] starting grpc/postgres server at cockroachdb-1.cockroachdb.production.svc.cluster.local:26257
I170510 17:35:21.840920 12 server/server.go:743  [n2] advertising CockroachDB node at cockroachdb-1.cockroachdb.production.svc.cluster.local:26257
F170510 17:35:22.081732 12 server/server.go:851  [n2] failed to run migration "create system.settings table": unexpected value: raw_bytes:"\306DY\334\001\014" timestamp:<wall_time:1494437673208752830 logical:0 > 
goroutine 12 [running]:
github.com/cockroachdb/cockroach/pkg/util/log.getStacks(0x0, 0x4df22e7, 0x2c014e0, 0x10)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:858 +0xa7
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).outputLogEntry(0x2c02cc0, 0xc400000004, 0x24c4521, 0x10, 0x353, 0xc420321d90, 0xa3)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:724 +0x591
github.com/cockroachdb/cockroach/pkg/util/log.addStructured(0x7f6898382dd8, 0xc42043e5d0, 0x4, 0x2, 0x0, 0x0, 0xc4207bb6a8, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/structured.go:158 +0x4b3
github.com/cockroachdb/cockroach/pkg/util/log.logDepth(0x7f6898382dd8, 0xc42043e5d0, 0x1, 0x4, 0x0, 0x0, 0xc4207bb6a8, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:68 +0x8c
github.com/cockroachdb/cockroach/pkg/util/log.Fatal(0x7f6898382dd8, 0xc42043e5d0, 0xc4207bb6a8, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:171 +0x76
github.com/cockroachdb/cockroach/pkg/server.(*Server).Start(0xc42061cd80, 0x7f6898382dd8, 0xc42043e5d0, 0xc4201f22c0, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:851 +0x2133
github.com/cockroachdb/cockroach/pkg/cli.runStart.func1.1(0x7f6898382dd8, 0xc420184f90, 0xc42000e2a8, 0xc42063d860, 0xc42015b320, 0x0, 0x6)
	/go/src/github.com/cockroachdb/cockroach/pkg/cli/start.go:357 +0x210
github.com/cockroachdb/cockroach/pkg/cli.runStart.func1(0x7f6898382dd8, 0xc420184f90, 0x27d49c0, 0x2c423e8, 0xc42000e2a8, 0xc42063d860, 0xc42015b320, 0xc4206a80c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/cli/start.go:414 +0xdf
created by github.com/cockroachdb/cockroach/pkg/cli.runStart
	/go/src/github.com/cockroachdb/cockroach/pkg/cli/start.go:417 +0x3b6

```

**Comment by tamird** [comment]
This is now (better) tracked in #16206.
